### PR TITLE
Feature log levels

### PR DESCRIPTION
**Pull request by rderimay**
Fixes Feature #71.
As I have seen that all the branches have been merged yesterday, I went the way to make one step only and replacing the old log call in the call with the new ones.

### CODE REVIEW

File: focuspoints.lrdevplugin/Utils.lua [hunk 1]
```diff
@@ -84,15 +85,70 @@ function stringToKeyValue(str, delim)
...
+  local msgLevel = levels[level]
+
+  if prefsLevel == 0 or msgLevel == nil or msgLevel > prefsLevel then
+    -- Unknown message log level or level set in preferences higher
+    -- No need to log this one, return
+    return
+  end
+
+  local str = string.format("%20s | %s", string.upper(string.sub(group, 1, 20)), message)
+  if level == "ERROR" then
```
Comment (musselwhizzle) [inline focuspoints.lrdevplugin/Utils.lua]: I accepted the PR, but would like to see this changed. This logs ONLY the set log level which is not the way any loggers work. If i have a log level of WARN, that means WARN and ERROR should log but not info. As I posted in the sample code, ``` if (level >= WARN) log. ```

Comment (rderimay) [inline focuspoints.lrdevplugin/Utils.lua]: No, it logs all levels which are of higher importance as the one selected in the prefs.

If you choose **WARN**, only WARN and ERROR levels will be logged, DEBUG and INFO will be ignored.
If you choose **INFO**, INFO, WARN and ERROR levels will be logged, DEBUG will be ignored.

Comment (musselwhizzle) [inline focuspoints.lrdevplugin/Utils.lua]: how are you doing that then? I'm seeing equality statements instead of >=

Comment (rderimay) [inline focuspoints.lrdevplugin/Utils.lua]: Using the levels table on the top of the fonction,
prefsLevel contains the corresponding **numeric** value of the level chosen in the prefs.
msgLevel contains the corresponding **numeric** value of the level of the message.

and then, Line 137:
 if prefsLevel == 0 or msgLevel == nil or msgLevel > prefsLevel then
    -- Unknown message log level or level set in preferences higher
    -- No need to log this one, return
    return
  end

Comment (musselwhizzle) [inline focuspoints.lrdevplugin/Utils.lua]: oh, you're using this as just a switch for which method to call and no filtering. filtering happens above and then you exit out.

Comment (rderimay) [inline focuspoints.lrdevplugin/Utils.lua]: yes. 
I would have loved to eliminate this
````
  if level == "ERROR" then
    myLogger:error(str)
  elseif level == "WARN" then
    myLogger:warn(str)
  elseif level == "INFO" then
    myLogger:info(str)
  else
    myLogger:debug(str)
  end
````
by something like
````
local methodName = "warn"
myLogger:methodName(str)
````
but I did not find a way to make it.

In php, that would have been:
````
$a = warn;
myLogger:$$a($str);
````

**Pull request merged**